lib/matplotlib/tests/test_figure.py

```diff
@@ -533,3 +533,202 @@ def test_removed_axis():
     fig, axs = plt.subplots(2, sharex=True)
     axs[0].remove()
     fig.canvas.draw()
+
+
+class TestCollage:
+    @check_figures_equal(extensions=['png'])
+    @pytest.mark.parametrize('x', [
+        [["A", "A", "B"], ["C", "D", "B"]],
+        [[1, 1, 2], [3, 4, 2]]]
+    )
+    def test_basic(self, fig_test, fig_ref, x):
+        grid_axes = fig_test.build_grid(x)
+
+        for k, ax in grid_axes.items():
+            ax.set_title(k)
+
+        labels = sorted(np.unique(x))
+
+        assert len(labels) == len(grid_axes)
+
+        gs = fig_ref.add_gridspec(2, 3)
+        axA = fig_ref.add_subplot(gs[:1, :2])
+        axA.set_title(labels[0])
+
+        axB = fig_ref.add_subplot(gs[:, 2])
+        axB.set_title(labels[1])
+
+        axC = fig_ref.add_subplot(gs[1, 0])
+        axC.set_title(labels[2])
+
+        axD = fig_ref.add_subplot(gs[1, 1])
+        axD.set_title(labels[3])
+
+    @check_figures_equal(tol=0.005, extensions=['png'])
+    def test_all_nested(self, fig_test, fig_ref):
+        x = [["A", "B"], ["C", "D"]]
+        y = [["E", "F"], ["G", "H"]]
+
+        grid_axes = fig_test.build_grid([[x, y]])
+        for ax in grid_axes.values():
+            ax.set_title(ax.get_label())
+
+        gs = fig_ref.add_gridspec(1, 2)
+        gs_left = gs[0, 0].subgridspec(2, 2)
+        for j, r in enumerate(x):
+            for k, label in enumerate(r):
+                fig_ref.add_subplot(gs_left[j, k]).set_title(label)
+
+        gs_right = gs[0, 1].subgridspec(2, 2)
+        for j, r in enumerate(y):
+            for k, label in enumerate(r):
+                fig_ref.add_subplot(gs_right[j, k]).set_title(label)
+
+    @check_figures_equal(tol=0.005, extensions=['png'])
+    def test_nested(self, fig_test, fig_ref):
+        x = [["A", "B", "B"], ["C", "C", "D"]]
+
+        y = [["F"], [x]]
+
+        grid_axes = fig_test.build_grid(y)
+
+        for k, ax in grid_axes.items():
+            ax.set_title(k)
+
+        gs = fig_ref.add_gridspec(2, 1)
+
+        gs_n = gs[1, 0].subgridspec(2, 3)
+
+        axA = fig_ref.add_subplot(gs_n[0, 0])
+        axA.set_title("A")
+
+        axB = fig_ref.add_subplot(gs_n[0, 1:])
+        axB.set_title("B")
+
+        axC = fig_ref.add_subplot(gs_n[1, :2])
+        axC.set_title("C")
+
+        axD = fig_ref.add_subplot(gs_n[1, 2])
+        axD.set_title("D")
+
+        axF = fig_ref.add_subplot(gs[0, 0])
+        axF.set_title("F")
+
+        fig_test.tight_layout()
+        fig_ref.tight_layout()
+
+    @check_figures_equal(tol=0.005, extensions=['png'])
+    def test_nested_tuple(self, fig_test, fig_ref):
+        x = [["A", "B", "B"], ["C", "C", "D"]]
+        xt = (("A", "B", "B"), ("C", "C", "D"))
+
+        fig_ref.build_grid([["F"], [x]])
+        fig_test.build_grid([["F"], [xt]])
+
+    @check_figures_equal(extensions=['png'])
+    @pytest.mark.parametrize('x, empty_sentinel', [
+        ([["A", None], [None, "B"]], None),
+        ([["A", ' '], [' ', "B"]], "SKIP"),
```

Comment:
please pick one style of quotes :)

Reply:
I mean if you are going to rebel against the automatic formatters....

(will push a fix soon).